x-pack/legacy/plugins/infra/public/containers/logs/log_position/log_position_state.ts

```diff
@@ -79,26 +111,68 @@ export const useLogPositionState: () => LogPositionStateParams & LogPositionCall
     [startKey, endKey]
   );
 
+  // Allow setting `startDate` and `endDate` separately, or together
+  const updateDateRange = useCallback(
+    (newDateRange: Partial<DateRange>) => {
+      // Prevent unnecessary re-renders
+      if (!('startDate' in newDateRange) && !('endDate' in newDateRange)) {
+        return;
+      }
+      if (
+        newDateRange.startDate === dateRange.startDate &&
+        newDateRange.endDate === dateRange.endDate
+      ) {
+        return;
+      }
+
+      jumpToTargetPosition(null);
+      setDateRange(previousDateRange => {
+        return {
+          startDate: newDateRange.startDate || previousDateRange.startDate,
+          endDate: newDateRange.endDate || previousDateRange.endDate,
+        };
+      });
+    },
+    [dateRange]
+  );
+
+  const startTimestamp = useMemo(() => datemathToEpochMillis(dateRange.startDate), [
```

Comment:
We keep `startTimestamp` and `endTimestamp` in `LogPositionState` to ensure all consumers always use the same timestamps for the same values of `*Date`. On an initial implementation I only exported the `*Date` values and let the consumers parse those into timestamps, but when using relative dates, different parts of the app got different timestamps and that caused problems.